cloud/services/loadbalancers.go

```diff
@@ -35,15 +37,69 @@ func EnsureNodeBalancer(ctx context.Context, clusterScope *scope.ClusterScope, l
 	}
 
 	logger.Info(fmt.Sprintf("Creating NodeBalancer %s", clusterScope.LinodeCluster.Name))
+
 	createConfig := linodego.NodeBalancerCreateOptions{
 		Label:  util.Pointer(clusterScope.LinodeCluster.Name),
 		Region: clusterScope.LinodeCluster.Spec.Region,
 		Tags:   []string{string(clusterScope.LinodeCluster.UID)},
 	}
 
+	// get firewall ID from firewallRef if it exists
+	if clusterScope.LinodeCluster.Spec.NodeBalancerFirewallRef != nil {
```

Comment:
Can we add it as a condition for linodecluster like we did for VPC?
https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodecluster_controller.go#L172-L181
https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodemachine_controller.go#L250-L258

Reply:
ah nice. I was wondering if I should do it like that. I'll switch it

Reply:
Done!